### PR TITLE
solve grid mean equations explicitly 

### PR DESCRIPTION
**Pull request by yairchn**

### CODE REVIEW

File: integration_tests/utils/generate_namelist.jl [hunk 1]
```diff
@@ -351,11 +351,11 @@ function SP(namelist_defaults)
     namelist["meta"]["casename"] = "SP"
 
     # this case is resolution dependent, we should check why
-    namelist["grid"]["nz"] = 256
-    namelist["grid"]["dz"] = 8
+    namelist["grid"]["nz"] = 128
```
Comment (yairchn) [inline integration_tests/utils/generate_namelist.jl:354]: SP does not work with dt=1 I had to make dz different. Change in the resolution effect SP, I have seen it before - I think something is off with the way the case is coded

File: integration_tests/utils/generate_namelist.jl [hunk 2]
```diff
@@ -128,7 +128,7 @@ function default_namelist(case_name::String)
     namelist_defaults["thermodynamics"]["quadrature_type"] = "log-normal" #"gaussian" or "log-normal"
 
     namelist_defaults["time_stepping"] = Dict()
-    namelist_defaults["time_stepping"]["dt"] = 10.0
+    namelist_defaults["time_stepping"]["dt"] = 7.0
```
Comment (yairchn) [inline integration_tests/utils/generate_namelist.jl:131]: at default dt =10 some minor instabilities still appear in diffusive cases such as dycoms and in the lower (boundary layer) of ARM

File: src/Turbulence_PrognosticTKE.jl [hunk 3]
```diff
@@ -173,14 +173,14 @@ function io(edmf::EDMF_PrognosticTKE, Stats::NetCDFIO_Stats, TS::TimeStepping)
     write_profile(Stats, "massflux_qt", mf_qt)
     write_profile(Stats, "massflux_tendency_h", edmf.massflux_tendency_h)
     write_profile(Stats, "massflux_tendency_qt", edmf.massflux_tendency_qt)
-    write_profile(Stats, "diffusive_flux_h", edmf.diffusive_flux_h)
-    write_profile(Stats, "diffusive_flux_qt", edmf.diffusive_flux_qt)
-    write_profile(Stats, "diffusive_flux_u", edmf.diffusive_flux_u)
-    write_profile(Stats, "diffusive_flux_v", edmf.diffusive_flux_v)
+    # write_profile(Stats, "diffusive_flux_h", edmf.diffusive_flux_h)
```
Comment (yairchn) [inline src/Turbulence_PrognosticTKE.jl:176]: we could either remove these diagnostics of handle them properly - depending on what do you think?

File: src/Turbulence_PrognosticTKE.jl [hunk 4]
```diff
@@ -674,27 +668,25 @@ function update(edmf::EDMF_PrognosticTKE, GMV::GridMeanVariables, Case::CasesBas
...
 end
 
-function update_GMV_turbulence(edmf::EDMF_PrognosticTKE, GMV::GridMeanVariables, Case::CasesBase, TS::TimeStepping)
-
-    @inbounds for k in real_center_indices(edmf.grid)
-        GMV.H.tendencies[k] += (GMV.H.new[k] - GMV.H.values[k]) * TS.dti
-        GMV.QT.tendencies[k] += (GMV.QT.new[k] - GMV.QT.values[k]) * TS.dti
-        GMV.U.tendencies[k] += (GMV.U.new[k] - GMV.U.values[k]) * TS.dti
-        GMV.V.tendencies[k] += (GMV.V.new[k] - GMV.V.values[k]) * TS.dti
+function GMV_update(grid::Grid, GMV::GridMeanVariables, TS::TimeStepping)
```
Comment (yairchn) [inline src/Turbulence_PrognosticTKE.jl:679]: I move 'update' from 'src/Variables.jl' to here - I believe it is more correct for this function to be here - in the future we could collated all update functions (up, en, gm) into a single function

File: src/types.jl [hunk 5]
```diff
@@ -238,12 +237,11 @@ struct VariablePrognostic{T}
         # Value at the current timestep
         values = field(grid, loc)
         # Value at the next timestep, used for calculating turbulence tendencies
-        new = field(grid, loc)
         tendencies = field(grid, loc)
         if kind != "scalar" && kind != "velocity"
             print("Invalid kind setting for variable! Must be scalar or velocity")
         end
-        return new{typeof(values)}(values, new, tendencies, loc, bc, kind, name, units)
+        return new{typeof(values)}(values, tendencies, loc, bc, kind, name, units)
```
Comment (yairchn) [inline src/types.jl:244]: no more 'new' in GMV!

**Review by charleskawczynski — COMMENTED** [review]
It looks like some of the results have degraded (e.g., some pretty strong oscillations)

**Comment by yairchn** [comment]
> It looks like some of the results have degraded (e.g., some pretty strong oscillations)

yes it was time-stepping , I updated the time steps lets see how it looks now.

File: src/Operators.jl [hunk 6]
```diff
@@ -36,6 +36,26 @@ end
...
+        return ∇c2f(f_dual, grid, TopBCTag(), top)
+    else
+        return ∇c2f(f_dual, grid, InteriorTag())
+    end
+end
+∇c2f(f::SA.SVector, grid::Grid, ::Int, ::UseBoundaryValue, top::UseBoundaryValue) = ∇c2f(f, grid, InteriorTag())
+∇c2f(f::SA.SVector, grid::Grid, ::InteriorTag) = (f[2] - f[1]) * grid.Δzi
+∇c2f(f::SA.SVector, grid::Grid, ::TopBCTag, bc::SetValue) = (bc.value - f[1]) * grid.Δzi * 2.0
+∇c2f(f::SA.SVector, grid::Grid, ::TopBCTag, bc::SetGradient) = bc.value
+∇c2f(f::SA.SVector, grid::Grid, ::BottomBCTag, bc::SetValue) = (f[2] - bc.value) * grid.Δzi * 2.0
```
Comment (charleskawczynski) [inline src/Operators.jl:55]:
```suggestion
∇c2f(f::SA.SVector, grid::Grid, ::BottomBCTag, bc::SetValue) = (f[1] - bc.value) * grid.Δzi * 2.0
```

**Review by charleskawczynski — APPROVED** [review]
Bomex / life cycle results are now looking better with the smaller timestep.

**Review by ilopezgp — APPROVED** [review]

**Pull request merged**